openml-generic-python/src/test/resources/valid_classifier/classifier.py

```diff
@@ -1,19 +1,25 @@
 import sys,math
 from ClassifierApi.classifier import ClassifierBase
+import random
+import numpy
 
 class Classifier(ClassifierBase):
 
     num_features=4
     num_classes=3
 
+    def normalize(self, v):
+        norm = numpy.linalg.norm(v, ord=1)
+        if norm == 0:
+            norm = numpy.finfo(v.dtype).eps
+        return v / norm
+
     def classify_instance(self, instance):
         return int(sum(instance) % self.num_classes)
 
     def getClassDistribution_instance(self, instance):
```

Comment:
note for the future: while it's resources code (used in tests), we should strive to keep it readable (and maintainable), as we will surely need to change this code in the future. This may imply adding comments if necessary to explain the logic (which in this case is not obvious IMHO).

So, to keep some record, I'm letting the explanation here in this PR comment for further reference: 

* old classifier version was returning always the same array for class distribution (`[0.2, 0.2, 0.6]`)   
* this new version now returns different values for each instance. to achieve that, it uses the first instance value as the random seed for the random generator. This will make the classDist deterministic for each instance, and somehow different across different instances (actually instances with the same first value will have the same classDist - we could have used the full array - but that's acceptable for now)

1 question nevertheless: is this `random` variable global? as in, if I now start calling `random.sample` in other methods, will that affect the classDist output because the internal state has changed? (this won't be a problem since we are not asserting the exact class distribution for an instance, it's just a curiosity)

Reply:
yes, it is a global variable inside the model. it may be a problem because even if we are not checking the exact results. we are checking that they have the same frequency (we are injecting 8 events with 2 distinct values, and so there should be also 2 distinct results each one with a frequency of 4).